base_location/models/res_company.py

```diff
@@ -48,24 +47,22 @@ def _get_company_address_field_names(self):
         return res
 
     def _inverse_city_id(self):
+        self = self.with_context(skip_check_zip=True)
         for company in self:
-            company.with_context(
-                skip_check_zip=True
-            ).partner_id.city_id = company.city_id
+            company.partner_id.city_id = company.city_id
 
     def _inverse_zip_id(self):
+        self = self.with_context(skip_check_zip=True)
         for company in self:
-            company.with_context(skip_check_zip=True).partner_id.zip_id = company.zip_id
+            company.partner_id.zip_id = company.zip_id
 
     def _inverse_state(self):
-        return super(
-            ResCompany, self.with_context(skip_check_zip=True)
-        )._inverse_state()
+        self = self.with_context(skip_check_zip=True)
```

Comment:
This is not correct, as it's a local variable and it's not pass upstream. Please revert.

Reply:
It is passed :)

Reply:
Well, I prefer not to do such things that depends on the Python implementation. Python zen: explicit is better than implicit.

Reply:
IMO it's explicit but ok, it's changed